datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -3397,6 +3462,75 @@ mod tests {
         // TODO: add test for other case and op
     }
 
+    #[test]
+    fn test_rewrite_expr_to_prunable_custom_unhandled_hook() {
+        struct CustomUnhandledHook;
+
+        impl UnhandledPredicateHook for CustomUnhandledHook {
+            /// This handles an arbitrary case of a column that doesn't exist in the schema
+            /// by renaming it to yet another column that doesn't exist in the schema
+            /// (the transformation is arbitrary, the point is that it can do whatever it wants)
+            fn handle(&self, _expr: &Arc<dyn PhysicalExpr>) -> Arc<dyn PhysicalExpr> {
+                Arc::new(phys_expr::Literal::new(ScalarValue::Int32(Some(42))))
+            }
+        }
+
+        let schema = Schema::new(vec![Field::new("a", DataType::Int32, true)]);
+        let schema_with_b = Schema::new(vec![
+            Field::new("a", DataType::Int32, true),
+            Field::new("b", DataType::Int32, true),
+        ]);
+
+        let transform_expr = |expr| {
+            let expr = logical2physical(&expr, &schema_with_b);
+            rewrite_predicate_to_statistics_predicate(
+                &expr,
+                &schema,
+                Some(Arc::new(CustomUnhandledHook {})),
+            )
+        };
+
+        // transform an arbitrary valid expression that we know is handled
+        let known_expression = col("a").eq(lit(ScalarValue::Int32(Some(12))));
+        let known_expression_transformed = rewrite_predicate_to_statistics_predicate(
+            &logical2physical(&known_expression, &schema),
+            &schema,
+            None,
+        );
+
+        // an expression referencing an unknown column (that is not in the schema) gets passed to the hook
+        let input = col("b").eq(lit(ScalarValue::Int32(Some(12))));
```

Comment:
I think you can write literals more concisely like this if you wanted

```suggestion
        let input = col("b").eq(lit(12i32));
```